### DISCUSSION TITLE
How to upload file (non-image) from web?

### DISCUSSION THREAD

**Discussion by sulthanalihsan** [serverpod/serverpod#962] [Q&A] [answered]
I'm stuck on uploading small files (non-image) especially on the web, I've tried several ways but still failed.

what is wrong with my code below?

```
final result = await FilePicker.platform
                        .pickFiles(type: FileType.any, allowMultiple: false);
                    if (result != null && result.files.isNotEmpty) {
                      final fileBytes = result.files.first.bytes;
                      final fileName = result.files.first.name;
                      var data = ByteData.view(fileBytes!.buffer);

                      var uploadDescription = await client.permohonan
                          .getUploadDescription('serverpod/file/$fileName');
                      if (uploadDescription != null) {
                        var uploader = FileUploader(uploadDescription);
                        await uploader.uploadByteData(data);
                        var success = await client.permohonan
                            .verifyUpload('serverpod/file/$fileName');
                      }
                    } 
```
the result (with no byte data)
![image](https://github.com/serverpod/serverpod/assets/30169687/fc7c722b-35e3-4299-8374-1aeedce158b4)

has anyone successfully uploaded file?
thankyou

**Comment by vlidholt** [comment, marked as answer]
Hi! Are you using a bucket (S3 or GCP Storage) or the built in database file storage? Unfortunately, the database file storage is not supported on web, but works on S3. Typically, you only want to store files in the database during development as Postgres isn't really built to handle many large files.

You can read more about the process of uploading files here: https://docs.serverpod.dev/concepts/file-uploads

**Reply by sulthanalihsan** [reply]
thanks for the answer, 

I use a database to store it and it turns out that it cannot store large files, my next question is whether it is possible to upload files to s3 compatible storage other than aws, because I have tried serverpod_cloud_storage_s3 to other providers but it still fails,

is there a way or some kind of modification or I need to use an external package?
